facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -423,7 +423,9 @@ public SSHCredentials getSshCredentials(String buildId) {
     @Override
     public Graph<Build> getBuildGraphForGroupBuild(String groupBuildId) {
         List<String> runningAndStoredIds = getBuildIdsInTheGroup(groupBuildId);
-
+        if (runningAndStoredIds.isEmpty()) {
+            throw new EmptyEntityException("Build group " + groupBuildId + " does not exists or has no builds.");
```

Comment:
We should return 404 when group build does not exists and empty graph when there are no builds in existing group build.

Reply:
https://github.com/project-ncl/pnc/pull/3246